### ISSUE TITLE
Append action erring on string with regex

### ISSUE THREAD

**Issue by jwrudzin** [plopjs/plop#188]
I'm seeing the same issue that was marked closed in [#178](https://github.com/plopjs/plop/issues/178). Opening a new issue here to track the problem.

I'm running plop v2.5.3 and seeing the same issue (using the same code) as outlined in #178.

**Comment by cakirlarc** [comment]
Similar regex problem while using append.
plop cant find the below string and appends again. The problem is with the "(" and ")". They must be escaped while regexing before append by plop.

`<Route key="AboutPage" exact path={PATHS.ABOUT_PAGE} render={(props) => <AboutPage {...props} />} />`

The

**Comment by cakirlarc** [comment]
the problem from@jwrudzin seems to be the "[" and "\" which must be escaped by plop to match

`{handle: "[data-{{ dashCase name}}]", require: {{properCase name}} }\,`

the correct regex to match the above is

`{handle: "\[data-{{ dashCase name}}]", require: {{properCase name}} }\\,`

 per https://regexr.com/

**Comment by cakirlarc** [comment]
in file append.js I made the below modification, and it seems to be working now for append. Someone may commit this maybe, I am not a familiar with source control systems.
````
    const stringToAppendSanitized = (separator + stringToAppend).replace(/[.*+?^${}()|[\]\\]/g, "\\$&") //added by cakirlarc
    // const lastPartWithoutDuplicates = lastPart.replace(new RegExp(separator + stringToAppend, 'g'), ''); //original plop
    const lastPartWithoutDuplicates = lastPart.replace(new RegExp(stringToAppendSanitized, 'g'), ''); //added by cakirlarc

````
@jwrudzin @amwmedia

**Comment by amwmedia** [comment]
Thanks, I'll take a look at this soon

**Comment by jwrudzin** [comment]
Has there been any update on this? I'm unfortunately still running into this issue even using the escaped regex string cakirlarc posted.

**Comment by crutchcorn** [comment]
I'll take a look at this tonight. I'll see what I can do about writing a test for this scenario as well

**Comment by crutchcorn** [comment]
Looking further, there seems to have been a PR already opened for this exact issue:

https://github.com/plopjs/node-plop/pull/139

We're waiting on some tests being written but once this is done, we can merge and cut a new release

**Comment by bryantAXS** [comment]
Looks like those tests were written -- any chance we can get that merged?

**Comment by crutchcorn** [comment]
@bryantAXS I don't seem to see tests in the pull request in question. Am I missing something obvious?

**Comment by crutchcorn** [comment]
Can someone give me an input that still throws this error? I am working on writing tests so that we can get this fix in (branch `fix-append-unique`) and get get `node-plop` to throw errors for either inputs:

```javascript
<Route key="AboutPage" exact path={PATHS.ABOUT_PAGE} render={(props) => <AboutPage {...props} />} />
```

Nor

```
{handle: "[data-{{ dashCase name}}]", require: {{properCase name}} }\,
```

Tests are a good thing to get working, but useless if I can't make them fail in the un-regexed codebase

**Comment by bryantAXS** [comment]
@crutchcorn sorry, I misread the pull request!

I just pulled down the latest `2.5.4` and it looks like I'm still running into the issue.

In this case the input used in the name variable is `TestingForGithub`

```
actions.push({
    type: 'append',
    path: 'src/scripts/app.js',
    pattern: /(Components Loader -- DO NOT REMOVE COMMENT!)/gi, 
    template: '{handle: "[data-{{ dashCase name}}]", require: {{properCase name}} }\,',
});
```

The errors generated are:

```
✔  ++ /src/components/molecules/TestingForGithub/TestingForGithub.js
✔  _+ /src/scripts/app.js
✖  _+ Invalid regular expression: /
{handle: "[data-testing-for-github]", require: TestingForGithub },/: Range out of order in character class
✖  ++ src/components/{{lowerCase type}}/{{properCase name}}/{{properCase name}}.scss Aborted due to previous action failure
✖  ++ src/components/{{lowerCase type}}/{{properCase name}}/{{properCase name}}.twig Aborted due to previous action failure
✖  _+ src/styles/includes/{{lowerCase type}}.scss Aborted due to previous action failure
```

Does that help?

**Comment by crutchcorn** [comment]
I'm still unable to get a failing test in `node-plop` for whatever reason when trying to write tests. I'll be taking a closer look at the integration within `plop` to see more specifically where the problem lies and if the PR that's been opened will really fix it. If it does, we'll merge. Just trying to do due diligence

**Comment by bryantAXS** [comment]
@crutchcorn sounds good - thanks!

**Comment by datner** [comment]
this can be worked around if you use the SafeString from handlebars, as such
```js
template: new Handlebars.SafeString('special string').toString()
```

at least in my case it solved the issue, but this is very cumbersome

**Comment by jwrudzin** [comment]
@crutchcorn Just checking to see if there's any update here? Still hoping to get this fix merged in. Thanks!

**Comment by crutchcorn** [comment]
I think we can merge in a PR that I have open for this https://github.com/plopjs/node-plop/pull/164. That said, I'd like someone to test that this actually solves the problem. If someone can pull in `node-plop` and `plop` and do some testing on a scenario they know is broken, then I can merge and we can cut a release

**Comment by novli** [comment]
The initial issue to be solved by https://github.com/plopjs/node-plop/pull/164 can be reproduced here https://codesandbox.io/s/ecstatic-dubinsky-igwz6?file=/plopfile.js
Just run `npx plop` with same name three times and open the index.ts file.

**Comment by valgussev** [comment]
any updates on this one?

**Comment by tomdev10** [comment]
Seeing the same error on `^2.7.4` with a `[` character in the template string.

Tried `new Handlebars.SafeString` as above, but same result.

`\\` to escape the characters removes the error, but leaves a single `\` in the appended content. Any help would be much appreciated 💪

**Comment by siosio34** [comment]
any updates on this one? this occur plop 3.1.1 v

**Comment by gustaff-weldon** [comment]
@siosio34 It does not seem so, the PR was moved to https://github.com/plopjs/plop/pull/330 from what I was able to find

**Comment by crtkl-mdoll** [comment]
It would be great to get https://github.com/plopjs/plop/pull/330 merged in for the next release. I spent way too long trying to troubleshoot why a template with "()" in it wasn't appending unique.
